apps/prairielearn/python/traverse.py

```diff
@@ -0,0 +1,183 @@
+import timeit
+from collections import deque
+from html import escape as html_escape
+from typing import Callable, Deque, List, Optional, Tuple, Union
+
+import lxml.html
+
+ElementReplacement = Optional[
+    Union[str, lxml.html.HtmlElement, List[lxml.html.HtmlElement]]
+]
+
+# https://developer.mozilla.org/en-US/docs/Glossary/Void_element
+VOID_ELEMENTS = {
```

Comment:
```suggestion
VOID_ELEMENTS = frozenset({
```
Similar comment as somewhere else, makes sense to declare this immutable.

Reply:
Done in 8e121e4. It's a little unfortunate how Black formats this!